src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -370,24 +386,57 @@ def action_version_create(
                 )
             )
             bundle_map = self._bundle()
-            version, patch = find_version_info_in_manifest_file(self.deploy_root)
-            if not version:
+            version_info = find_version_info_in_manifest_file(self.deploy_root)
+            version_resolved, patch_manifest, label_manifest = (
+                version_info.version_name,
+                version_info.patch_number,
+                version_info.label,
+            )
+            if version_resolved is None:
                 raise ClickException(
                     "Manifest.yml file does not contain a value for the version field."
                 )
 
-        # Check if --patch needs to throw a bad option error, either if application package does not exist or if version does not exist
-        if patch is not None:
+            # If patch is set in CLI and is also present in manifest.yml with different value, confirmation from user is required to ignore patch from manifest.yml and proceed with CLI value.
+            if (
+                patch is not None
+                and patch_manifest is not None
+                and patch_manifest != patch
+            ):
+                console.warning(
+                    f"Cannot resolve version. Found patch: {patch_manifest} in manifest.yml which is different from provided patch {patch}. Re-run command with --force to ignore patch in manifest.yml"
```

Comment:
Not sure we should instruct the user to re-run unless we're in non-interactive mode.